mlir/lib/Dialect/Tosa/IR/TosaCanonicalizations.cpp

```diff
@@ -109,6 +109,102 @@ void ConcatOp::getCanonicalizationPatterns(RewritePatternSet &results,
   results.add<SelfConcatToTile>(context);
 }
 
+/* Rewrites reshapes that are adding 1-sized dims and are followed by a tile on
+ * the one-sized dim to a tile on the original shape followed by a reshape. This
+ * is done to reduce the rank of tile ops. */
```

Comment:
Is the core canonicalization to minimize the dimensions needed to express the `tile` operation? If so it would be better to do so regardless of a directly connected reshape or not and instead insert `reshape` before and after as appropriate.

If this is not the core goal then I question having this as a canonicalization as matching tile will still have a guarantee that the tile op is in canonicalized form.